File: bootstrap/src/test/java/net/dontdrinkandroot/wicket/bootstrap/TestApplication.java
package net.dontdrinkandroot.wicket.bootstrap;

import org.apache.wicket.Page;
import org.apache.wicket.protocol.http.WebApplication;


public class TestApplication extends WebApplication
{

	@Override
	public Class<? extends Page> getHomePage()
	{
		return TestHomePage.class;
	}

}

File: bootstrap/src/test/kotlin/net/dontdrinkandroot/wicket/bootstrap/test/TestApplication.kt
package net.dontdrinkandroot.wicket.bootstrap.test

import org.apache.wicket.Page
import org.apache.wicket.Session
import org.apache.wicket.protocol.http.WebApplication
import org.apache.wicket.protocol.http.WebSession
import org.apache.wicket.request.Request
import org.apache.wicket.request.Response
import java.util.*

fun getTestApplication(): TestApplication = WebApplication.get() as TestApplication

class TestApplication : WebApplication() {

    override fun getHomePage(): Class<out Page> = TestHomePage::class.java

    override fun newSession(request: Request, response: Response): Session {
        return object : WebSession(request) {
            override fun getLocale(): Locale {
                return Locale.ENGLISH
            }
        }
    }
}